### ISSUE TITLE
Some GUI apps (firefox, burpsuite, wireshark) cannot open display :0

### ISSUE THREAD

**Issue by HughRaynor** [microsoft/wslg#165]
# Environment
```
Windows buld number: Microsoft Windows [Version 10.0.21364.1011]
Your distribution version: Release: 2021.1
Your WSLg version: 1.0.19
```

# Steps to reproduce

Run one of a number of GUI apps including `firefox`, `burpsuite` or `wireshark`. Note that some apps, like Xfce or gedit, work as intended.

[weston.log](https://github.com/microsoft/wslg/files/6426499/weston.log)
[pulseaudio.log](https://github.com/microsoft/wslg/files/6426530/pulseaudio.log)

# Expected behavior

The application launches.

# Actual behavior

The following error is thrown:

```Unable to init server: Could not connect: Connection refused
Error: cannot open display: :0
```

# Additional information

```
┌──(hray㉿DESKTOP-HUGH)-[/tmp/.X11-unix]
└─$ env display                                                                                                   127 ⨯
display-im6.q16: unable to open X server `:0' @ error/display.c/DisplayImageCommand/412.
```

```
┌──(hray㉿DESKTOP-HUGH)-[/tmp/.X11-unix]
└─$  ls -la /tmp/.X11-unix                                                                                          1 ⨯
total 64
drwxrwxrwt  2 root root  4096 May  5 10:04 .
drwxrwxrwt 63 root root 57344 May  5 09:56 ..
srwxrwxrwx  1 hray hray     0 Mar 12 09:22 X10
srwxrwxrwx  1 hray hray     0 Mar 12 09:29 X11
lrwxrwxrwx  1 root root    19 May  5 10:04 .X11-unix -> /mnt/wslg/.X11-unix
srwxrwxrwx  1 hray hray     0 Mar 12 10:08 X12
srwxrwxrwx  1 hray hray     0 Mar 16 18:35 X13
srwxrwxrwx  1 hray hray     0 Mar 31 16:48 X14
```

```
┌──(hray㉿DESKTOP-HUGH)-[/tmp/.X11-unix]
└─$ ls -la /tmp/.X11-unix/X11                                                                                     130 ⨯
srwxrwxrwx 1 hray hray 0 Mar 12 09:29 /tmp/.X11-unix/X11
```

**Comment by spronovo** [comment]
Hi @HughRaynor, it looks like your ```/tmp/.X11-unix``` mapping is incorrect in the output you pasted above. In particular the mapping should be

```
/tmp/.X11-unix -> /mnt/wslg/.X11-unix
```

Above it looks like you have .X11-unix repeated as below

```
/tmp/.X11-unix/.X11-unix -> /mnt/wslg/.X11-unix
```

If you fix that, hopefully it should start working correctly and you should see an X0 socket for the WSLg X server.

```
/tmp/.X11-unix/X0
```

**Comment by HughRaynor** [comment]
Odd, I configured the mapping based on a comment you made on another issue.

This is what occurs when I try to reconfigure the mapping.

`ln -s /mnt/wslg/.X11-unix /tmp/.X11-unix`

**Comment by HughRaynor** [comment]
Resolved.

I had to perform the following actions.

```
┌──(hray㉿DESKTOP-HUGH)-[/tmp]
└─$ sudo rm -rf .X11-unix                                                                                                                                                                                   

┌──(hray㉿DESKTOP-HUGH)-[/tmp]
└─$ ln -s /mnt/wslg/.X11-unix /tmp/.X11-unix
```

**Comment by spronovo** [comment]
Hey @HughRaynor, your correct. In our [wiki](https://github.com/microsoft/wslg/wiki/Diagnosing-%22cannot-open-display%22-type-issues-with-WSLg) i had originally only included the ```ln -s /mnt/wslg/.X11-unix /tmp/.X11-unix``` command... until another user made me realize that some users weren't just missing the directory... but for them it had been replaced entirely... so the command would end up creating the incorrect nested link. A couple of days ago I updated the wiki to first delete the directory as you figured out above if it already exist such that the correct link is created :-).
